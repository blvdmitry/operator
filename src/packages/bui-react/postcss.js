// Re-export of the config provided by bui-postcss package
const config = require("@bookingcom/bui-postcss");

module.exports = config.default;
