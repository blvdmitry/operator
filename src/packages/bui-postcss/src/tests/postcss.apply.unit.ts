/* eslint-disable import/no-extraneous-dependencies */
import postcss from "postcss";

const postcssApply = require("../postcss.apply");

test("postcss.apply", async () => {
  const input = `
    .icon {
      @apply --bui_focus;
    }
    
    .heading {
      @apply (--bui_heading);
    }

    :root {
      --bui_focus {
        color: inherit;
      }
      --bui_heading: {
        font-size: 20px;
      }
    }
  `;
  const output = await postcss([postcssApply()]).process(input, {
    from: "__placeholder-name.css",
  });

  expect(output.css).toMatchSnapshot();
});
