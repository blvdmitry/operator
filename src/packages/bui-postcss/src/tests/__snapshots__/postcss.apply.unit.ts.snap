// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postcss.apply 1`] = `
"
    .icon {
      color: inherit;
    }
    
    .heading {
      font-size: 20px;
    }
  "
`;
